src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -944,7 +946,10 @@ def from_pretrained(cls, pretrained_model_or_path, **kwargs):
         if "enable_pag" in kwargs:
             enable_pag = kwargs.pop("enable_pag")
             if enable_pag:
-                orig_class_name = config["_class_name"].replace("Pipeline", "PAGPipeline")
+                if "controlnet" in kwargs:
```

Comment:
can you provide an example where this code addition is needed? I think it should already be addressed in line 944 no?

Reply:
Hi @yiyixuxu,

Thank you very much for your comments. Sorry if I am missing something here, but my idea is as follows:
There are 4 cases that need to be handled:

- No ControlNet and no Pag -> Keep **Pipeline**

- With ControlNet and no Pag -> Replace Pipeline with **ControlNetPipeline**

- No ControlNet and with Pag -> Replace Pipeline with **PAGPipeline**

- With ControlNet and with Pag -> Replace Pipeline with **ControlNetPAGPipeline**, which will override the first case.

That's why I added the nested conditions, but maybe I am missing something 🤔

Reply:
i see! thank for explaining it, you were absolute ly right! 


can we do this instead (to be consistent with AutoPipelineForText2Image and AutoPipelineForImage2Image)

```python
if "controlnet" in kwargs:
            orig_class_name = config["_class_name"].replace("Pipeline", "ControlNetPipeline")
        if "enable_pag" in kwargs:
            enable_pag = kwargs.pop("enable_pag")
            if enable_pag:
                orig_class_name = orig_class_name.replace("Pipeline", "PAGPipeline")
```
https://github.com/huggingface/diffusers/blob/73acebb8cfbd1d2954cabe1af4185f9994e61917/src/diffusers/pipelines/auto_pipeline.py#L363

Reply:
Oh sure, done! This way is better!